File: leetcode/easy/217-containsDuplicate.java
// first pass solution
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> set = new HashSet<>();
        boolean output = false;
        
        for (int n : nums) {
            if (set.contains(n)) {
                output = true;
                break;
            }
            set.add(n);
        }
        return output;
    }
}

// second pass solution
class Solution {
    public boolean containsDuplicate(int[] nums) {
        Arrays.sort(nums);
        
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] == nums[i-1]) return true;
        }
        return false;
    }
}

File: leetcode/easy/217-containsDuplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        set<int> set;
        int i = 0;
        bool flag = false;

        while (i < nums.size() && !flag) {
            if (set.find(nums[i]) != set.end()) {
                flag = true;
            }
            set.insert(nums[i]);
            i++;
        }

        return flag;
    }
};
